conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -108,6 +109,10 @@ public static JavaFile generateAliasType(
                 .addStatement("return new $T(value)", thisClass)
                 .build());
 
+        if (typeDef.getAlias().accept(TypeVisitor.IS_OPTIONAL)) {
+            spec.addMethod(createDefaultConstructor(aliasTypeName));
+        }
```

Comment:
Might be worth a comment explaining why we're adding a private no-arg constructor.